### DISCUSSION TITLE
v3.3.0

### DISCUSSION THREAD

**Discussion by rfelber** [secureCodeBox/secureCodeBox#780] [Releases]
## Changes
This release contains the following changes 🎉. Help spread the word or leave a GitHub star if you like it 😉
  
![GitHub commits since tagged version](https://img.shields.io/github/commits-since/secureCodeBox/secureCodeBox/v3.2.0) ![GitHub Repo stars](https://img.shields.io/github/stars/secureCodeBox/secureCodeBox?style=social) ![Twitter URL](https://img.shields.io/twitter/url?style=social&url=https%3A%2F%2Fgithub.com%2FsecureCodeBox%2FsecureCodeBox%2Freleases%2Ftag%2Flatest)

- Enable templating of environmental variables in cascading scans @malexmave (#773)

## 🚀 Features

- Added new SAST scanner `semgrep` @malexmave (#744)
- Added optional mode to run `DefectDojo` with a lower privileged user @J12934 (#747)
- Let AutoDiscovery create ScheduledScans with `RetriggerOnScanTypeChange` enabled @J12934 (#748)

## 🚓 Security Scanner

- [SCB-Bot] Upgraded `amass` from 3.14 to 3.14.1 @github-actions (#743)
- [SCB-Bot] Upgraded `amass` from 3.14.1 to 3.14.2 @github-actions (#776)
- [SCB-Bot] Upgraded `zap-advanced` from 2.10.0 to 2.11.0 @github-actions (#768)
- [SCB-Bot] Upgraded `zap` from 2.10.0 to 2.11.0 @github-actions (#767)
- [SCB-Bot] Upgraded `sslyze` from 4.0.4 to 4.1.0 @github-actions (#764)
- [SCB-Bot] Upgraded `kube-hunter` from 0.6.1 to 0.6.3 @github-actions (#762)
- Added new SAST scanner `semgrep` @malexmave (#744)
- Git-Repo-Scanner: Optionally include commit information in output @malexmave (#775)

## ⚓️ Hooks

- Added initContainer support in cascading scan rules @malexmave (#734)

## 🐛 Bug Fixes

- Symlink license file into toplevel dir @Weltraumschaf (#774)
- Fixed Issue in Helm Release Process @J12934 (#753)
- Fixed incompatibility with BSD Unix sed in makefile (closes #749) @Weltraumschaf (#751)
- Fixed timezone issues with `git-repo-scanner` @malexmave (#745)

## 📚 Documentation

- Added a warning about `DefectDojo` deduplication issue to the documentation @malexmave (#760)

## 🔧 Maintenance

- Aligned common targets for SDK makefile @EndPositive (#758)

## 📌 Dependencies

- [SCB-Bot] Upgraded `amass` from 3.14 to 3.14.1 @github-actions (#743)
- [SCB-Bot] Upgraded `amass` from 3.14.1 to 3.14.2 @github-actions (#776)
- [Snyk] Upgrade axios from 0.21.4 to 0.22.0 @snyk-bot (#772)
- [SCB-Bot] Upgraded `zap-advanced` from 2.10.0 to 2.11.0 @github-actions (#768)
- [SCB-Bot] Upgraded `zap` from 2.10.0 to 2.11.0 @github-actions (#767)
- [SCB-Bot] Upgraded `sslyze` from 4.0.4 to 4.1.0 @github-actions (#764)
- [SCB-Bot] Upgraded `kube-hunter` from 0.6.1 to 0.6.3 @github-actions (#762)
- [Snyk] Upgrade axios from 0.21.4 to 0.22.0 @snyk-bot (#771)
- Enabled SCB-Bot scanner upgrades @rseedorff (#769)

## Distribution
[![Artifact HUB](https://img.shields.io/endpoint?url=https://artifacthub.io/badge/repository/secureCodeBox)](https://artifacthub.io/packages/search?repo=securecodebox)
[![Docker Hub](https://img.shields.io/badge/Docker%20Hub-secureCodeBox-blue?style=flat&logo=docker&color=lightblue&labelColor=blue&logoColor=white)](https://hub.docker.com/u/securecodebox) 

## Contributers
Thx to all our contributers supporting this project 🤗
@EndPositive, @J12934, @Weltraumschaf, @github-actions, @malexmave, @rseedorff, @secureCodeBoxBot and @snyk-bot

<hr /><em>This discussion was created from the release <a href='https://github.com/secureCodeBox/secureCodeBox/releases/tag/v3.3.0'>v3.3.0</a>.</em>

**Comment by malexmave** [comment]
This release, combined with the previous release, updated the custom resource definitions (CRDs) for the secureCodeBox. Due to [limitations with helm](https://helm.sh/docs/topics/charts/#limitations-on-crds), they are not updated automatically when you update the operator, you will have to run the following code against your Kubernetes cluster to update them manually:

```bash
kubectl apply -f https://raw.githubusercontent.com/secureCodeBox/secureCodeBox/v3.3.0/operator/crds/cascading.securecodebox.io_cascadingrules.yaml
kubectl apply -f https://raw.githubusercontent.com/secureCodeBox/secureCodeBox/v3.3.0/operator/crds/execution.securecodebox.io_parsedefinitions.yaml
kubectl apply -f https://raw.githubusercontent.com/secureCodeBox/secureCodeBox/v3.3.0/operator/crds/execution.securecodebox.io_scancompletionhooks.yaml
kubectl apply -f https://raw.githubusercontent.com/secureCodeBox/secureCodeBox/v3.3.0/operator/crds/execution.securecodebox.io_scans.yaml
kubectl apply -f https://raw.githubusercontent.com/secureCodeBox/secureCodeBox/v3.3.0/operator/crds/execution.securecodebox.io_scantypes.yaml
kubectl apply -f https://raw.githubusercontent.com/secureCodeBox/secureCodeBox/v3.3.0/operator/crds/execution.securecodebox.io_scheduledscans.yaml
```

If you do not do this, you will get [errors when trying to use initContainers](https://github.com/secureCodeBox/secureCodeBox/issues/784).
